### ISSUE TITLE
failed isinstance loses TypeVar information

### ISSUE THREAD

**Issue by shinnar** [microsoft/pyright#1221]
**Describe the bug**
When an isinstance check in a conditional narrows a generic type, the else branch loses information.

**To Reproduce**
a repro case is included below.

**Expected behavior**
The else branch should still know that step is an OpType (actually, the if branch should as well, but I don't know if that is supportable without intersection types)

**Screenshots or Code**
If applicable, add screenshots or the text of the code (surrounded by triple back ticks) to help explain your problem.
```python
from typing import List, TypeVar, Generic


class Operator:
    ...


OpType = TypeVar("OpType", bound=Operator)


class BasePipeline(Operator, Generic[OpType]):
    _steps: List[OpType]

    def __init__(
        self,
        steps: List[OpType],
    ) -> None:
        self._steps = []

        for step in steps:
            # This line works without a problem
            self._steps.append(step)
            if isinstance(step, BasePipeline):
                pass
            else:
                # This line triggers an error:
                # error: Argument of type "Operator" cannot be assigned to parameter "__object" of type "OpType" in function "append"
                # Type "Operator" cannot be assigned to type "OpType" (reportGeneralTypeIssues)
                self._steps.append(step)
                reveal_type(step) # Type of "step" is "Operator"
```

**VS Code extension or command-line**
pyright command line. `pyright --version` returns
`pyright 1.1.90`

**Additional context**
This case was cut down from an existing code base that is trying to use pyright

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is now addressed in pyright 1.1.91, which I just published. It will also be included in the next release of pylance.
